funnel/models/mixins.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+
+import urllib.parse
+
+import dateutil.parser
+import requests
+
+from coaster.utils import parse_duration
+
+from .. import app, redis_store
+from . import db
+
+
+class VideoMixin:
+    video_id = db.Column(db.UnicodeText, nullable=True)
+    video_source = db.Column(db.UnicodeText, nullable=True)
+
+    _video_duration = 0
+    _video_uploaded_at = None
+
+    @property
+    def video(self):
+        data = None
+        if self.video_source and self.video_id:
+            data = redis_store.hgetall(self.cache_key)
```

Comment:
Should be `self.video_cache_key` to avoid `cache_key` naming conflicts from subclasses.